### PR TITLE
[backend] implemented new workerstates dead/away

### PR DESCRIPTION
**Pull request by lethliel**
Introducing new workerstates

- **dead:** worker is not reachable or broken and cannot be used for building.
- **away:** worker is building for another reposerver.

### CODE REVIEW

File: src/backend/BSXML.pm [hunk 1]
```diff
@@ -734,6 +734,7 @@ our $worker = [
 	'hostarch',
 	'ip',
 	'port',
+        'reposerver',
```
Comment (mlschroe) [inline src/backend/BSXML.pm]: Oh my. We already use 'reposerver' for masterdispatched jobs, so we need to use some other name here. Sorry.

File: src/backend/bs_warden [hunk 2]
```diff
@@ -121,7 +121,7 @@ while (1) {
 	};
 	eval {
 	  BSRPC::rpc($param, undef, 'cmd=idleworker', "workerid=$worker->{'workerid'}", "jobid=$js->{'jobid'}");
-	  unlink("$workersdir/building/$wname");
```
Comment (mlschroe) [inline src/backend/bs_warden]: Why did you delete that line?

File: src/backend/BSXML.pm (hunk 1 above)
Comment (lethliel) [inline src/backend/BSXML.pm]: no problem. I will rename it, Can I use buildreposerver?

File: src/backend/bs_warden (hunk 2 above)
Comment (lethliel) [inline src/backend/bs_warden]: my bad. I wanted to add another unlink and deleted this one. Sorry.

**Comment by lethliel** [comment]
reviewed with Michael
